src/org/zaproxy/zap/extension/websocket/manualsend/ManualWebSocketSendEditorDialog.java

```diff
@@ -261,5 +273,72 @@ public void addSeparator() {
 		public void addEndButton(JButton button) {
 			messagePanel.addOptions(button, HttpPanel.OptionsLocation.END);
 		}
+
+		private JButton getBtnReopen(WebSocketPanelSender sender) {
+			if (btnReopen == null) {
+				btnReopen = new JButton();
+				btnReopen.setIcon(reopenIcon);
+				btnReopen.setEnabled(true);
+				btnReopen.setToolTipText(Constant.messages.getString("websocket.manual_send.btn_reopen_hint"));
+				btnReopen.addActionListener(e -> {
+					if(wsUiHelper.isAlwaysGenerate()){
+						newSecWebSocketKey = WebSocketUtils.generateSecWebSocketKey();
+					}else {
+						newSecWebSocketKey = ((wsUiHelper.isGenerateKeyPressed()) ? wsUiHelper.getWebSocketKey() : null);
+					}
+					if(!sender.reOpenChannel(wsUiHelper.getSelectedChannelId(), newSecWebSocketKey, wsUiHelper.isRedirection(), wsUiHelper.isTrackingSession())){
+						View.getSingleton().showWarningDialog(Constant.messages.getString("websocket.manual_send.fail.unable_reopen"));
+					}
+				});
+			}
+			return btnReopen;
+		}
+
+		private JButton getBtnReopenEdit(WebSocketPanelSender sender){
+			if(btnReopenEdit == null) {
+				btnReopenEdit = new JButton();
+				btnReopenEdit.setIcon(reopenEditIcon);
+				btnReopenEdit.setEnabled(true);
+				btnReopenEdit.setToolTipText(Constant.messages.getString("websocket.manual_send.btn_reopen_edit_hint"));
+				btnReopenEdit.addActionListener(e -> {
+					try {
+						wsUiHelper.setSecWebsocketKeyField(sender.getSecWebKeyFrom(wsUiHelper.getSelectedChannelId()));
+						JOptionPane.showMessageDialog(this, getReopenPanel(),
+								Constant.messages.getString("websocket.manual_send.adv_dialog.title"),
+								JOptionPane.PLAIN_MESSAGE);
+					} catch (IllegalArgumentException e1) {
+						View.getSingleton().showWarningDialog(Constant.messages.getString("websocket.manual_send.fail.invalid_channel"));
+					} catch (DatabaseException e1) {
+						e1.printStackTrace();
+					} catch (HttpMalformedHeaderException e1) {
+						e1.printStackTrace();
+					}
+				});
+			}
+			return btnReopenEdit;
+		}
+
+		private JPanel reopenPanel;
```

Comment:
I get the idea of declaring vars close to the point where they're used, but I think this class (and the majority of the code base) generally declare members/fields up front.